runtime/polkadot/src/xcm_config.rs

```diff
@@ -134,24 +134,31 @@ pub type XcmRouter = (
 
 parameter_types! {
 	pub const Dot: MultiAssetFilter = Wild(AllOf { fun: WildFungible, id: Concrete(TokenLocation::get()) });
-	pub const DotForStatemint: (MultiAssetFilter, MultiLocation) = (Dot::get(), Parachain(STATEMINT_ID).into_location());
-	pub const CollectivesLocation: MultiLocation = Parachain(COLLECTIVES_ID).into_location();
-	pub const DotForCollectives: (MultiAssetFilter, MultiLocation) = (Dot::get(), CollectivesLocation::get());
+	pub DotForStatemint: (MultiAssetFilter, MultiLocation) = (Dot::get(), Parachain(STATEMINT_ID).into_location());
+	pub CollectivesLocation: MultiLocation = Parachain(COLLECTIVES_ID).into_location();
+	pub DotForCollectives: (MultiAssetFilter, MultiLocation) = (Dot::get(), CollectivesLocation::get());
 	pub const MaxAssetsIntoHolding: u32 = 64;
 }
 
 /// Polkadot Relay recognizes/respects the Statemint chain as a teleporter.
 pub type TrustedTeleporters =
 	(xcm_builder::Case<DotForStatemint>, xcm_builder::Case<DotForCollectives>);
 
-match_types! {
-	pub type OnlyParachains: impl Contains<MultiLocation> = {
-		MultiLocation { parents: 0, interior: X1(Parachain(_)) }
-	};
-	pub type CollectivesOrFellows: impl Contains<MultiLocation> = {
-		MultiLocation { parents: 0, interior: X1(Parachain(COLLECTIVES_ID)) } |
-		MultiLocation { parents: 0, interior: X2(Parachain(COLLECTIVES_ID), Plurality { id: BodyId::Technical, .. }) }
-	};
+pub struct OnlyParachains;
+impl Contains<MultiLocation> for OnlyParachains {
+	fn contains(loc: &MultiLocation) -> bool {
+		matches!(loc.unpack(), (0, [Parachain(_)]))
+	}
+}
+
+pub struct CollectivesOrFellows;
+impl Contains<MultiLocation> for CollectivesOrFellows {
+	fn contains(loc: &MultiLocation) -> bool {
+		matches!(loc.unpack(), |(0, [Parachain(COLLECTIVES_ID)])| (
+			0,
+			[Parachain(COLLECTIVES_ID), Plurality { id: BodyId::Technical, .. }]
+		))
+	}
```

Comment:
I'd prefer to keep the declarative syntax. As a rule, top-level runtime files should not contain procedural code.